app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/widget/EditCardNameDialog.kt

```diff
@@ -0,0 +1,99 @@
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.OutlinedButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import io.snabble.pay.app.R
+import io.snabble.pay.app.ui.theme.SnabblePayTheme
+
+@Composable
+fun EditCardNameDialog(
+    cardName: String,
+    onDismissRequest: () -> Unit,
+    onSaveNameRequest: (String) -> Unit,
+) {
+    val editTextValue = rememberSaveable(inputs = arrayOf(cardName)) {
+        mutableStateOf(cardName)
+    }
+    Dialog(onDismissRequest = onDismissRequest) {
+        Surface(
+            shape = RoundedCornerShape(24.dp)
+        ) {
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(16.dp),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                Text(text = stringResource(id = R.string.account_details_change_title), style = MaterialTheme.typography.titleLarge)
+                Spacer(modifier = Modifier.size(16.dp))
+                EditTextFieldCentered(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(horizontal = 16.dp),
+                    value = editTextValue.value,
+                    onValueChange = { editTextValue.value = it },
+                    onAction = {
+                        onSaveNameRequest(editTextValue.value)
+                    }
+                )
+                Spacer(modifier = Modifier.size(16.dp))
+                OutlinedButton(
+                    modifier = Modifier
+                        .fillMaxWidth(),
+                    shape = RoundedCornerShape(32.dp),
+                    border = BorderStroke(1.dp, MaterialTheme.colorScheme.onSurface),
+                    colors = ButtonDefaults.outlinedButtonColors(
+                        backgroundColor = Color.Transparent
+                    ),
+                    onClick = {
+                        onSaveNameRequest(editTextValue.value)
+                    }
+                ) {
+                    Text(
+                        modifier = Modifier
+                            .padding(start = 8.dp),
+                        text = stringResource(id = R.string.change_title_dialog_button),
+                        style = MaterialTheme.typography.labelLarge
+                    )
+                }
+            }
+        }
+    }
```

Comment:
Maybe we can merge this Dialog and the Dialog from [here](https://github.com/snabble/snabble-pay-android-sdk/blob/82b2b93c92a6f426dc7edcd3e3d68c15ad29fcd4/app/src/main/java/io/snabble/pay/app/ui/SnabblePayDialog.kt#L44) to one implementation 👍

Reply:
I don't think we should merge everything simply bevause we can. The dialogs implement completly different usescase and thus shouldn't be merged to one.

Reply:
To be fair, it has nothing todo with we can and cannot do. It's about consistency throughout the app.

See, both dialogs as they have been implemented according to "requirements":
|#35|#37 _(this PR)_|
|--|--|
|<img width="321" alt="Screenshot 2024-01-22 at 09 05 21" src="https://github.com/snabble/snabble-pay-android-sdk/assets/1065776/1bd0b671-ceb5-4084-9aa2-1d921a0f79bd">|<img width="316" alt="Screenshot 2024-01-22 at 09 07 12" src="https://github.com/snabble/snabble-pay-android-sdk/assets/1065776/ef90fcba-f42f-462f-908d-040ea0d8ee1e">|

I don't think that this is what we really want. I'd say latest changes are a step towards the right direction about appearance of the dialogs:

|#35|#37 _(this PR)_|
|--|--|
|<img width="326" alt="Screenshot 2024-01-22 at 09 12 45" src="https://github.com/snabble/snabble-pay-android-sdk/assets/1065776/5fe519ff-850f-4a1f-a064-98f0864f3e19">|<img width="316" alt="Screenshot 2024-01-22 at 09 07 12" src="https://github.com/snabble/snabble-pay-android-sdk/assets/1065776/9a589768-1ec7-4c70-8b0b-657c96c2e4dc">|

What do you think?

Reply:
I simply say we should not merge the implementations together. We have a simply alert dialog with the functionality to accept or decline and a dialog to edit something. Completlly differen purposes wich has nothing to do with the otherall desing. 
If u refere to my old implemenation of the alert dialog and the design difference to the edit dialog, i agree with you. But these changes have been applied in the other PR. If u compare the last given images the design matches. Also this dialog has been approved as u can read in the comment section of the affected ticket.